services/net/content/ContentManager.cs

```diff
@@ -229,7 +229,8 @@ private async Task HandleMessageAsync(ConsumeResult<string, SourceContent> resul
                 // TODO: Failures after receiving the message from Kafka will result in missing content.  Need to handle this scenario.
                 // TODO: Handle e-tag.
                 var source = await _api.GetSourceForCodeAsync(model.Source);
-                if (model.ProductId == 0)
+                var productId = source.ProductId ?? model.ProductId;
```

Comment:
Do not put this here.  The ingest that found this content should be making this decision on what the valid `ProductId` should be.

Reply:
Fixed.